### ISSUE TITLE
Property name in GangMemberInfo and GangMemberAscension causes unnecessary RAM cost

### ISSUE THREAD

**Issue by cigarmemr** [bitburner-official/bitburner-src#875]
### Steps to reproduce
1. Write ns2 codes below:
```
ns.gang.getMemberInformation('').hack
ns.gang.getAscensionResult('').hack
```
2. Check RAM usage

### Result
RAM cost of function hack() is calculated:
![image](https://github.com/bitburner-official/bitburner-src/assets/26317456/fb471290-5df6-4f85-824e-45495dcdb58c)
which is irrelevant in gang managing scenairos.

### Game version
Bitburner v2.5.0 (b87b8b4be)

**Comment by Hoekstraa** [comment]
Similar issue to #298.

**Comment by cigarmemr** [comment]
> Similar issue to #298.

Indeed. I suppose the ultimate solution to similar issues to be changing either property identifiers (e.g. from `hack` to `hacking`), or modify the commonly used `hack()` `grow()` `weaken()` to something else like `toHack()` `toGrow()` `toWeaken()`.

Using index-style `object['hack']` compromises this but yet lacks simplicity and fails to utilize auto-complete feature in Script Editor.

**Comment by gmcew** [comment]
> I suppose the ultimate solution to similar issues to be changing either property identifiers (e.g. from `hack` to `hacking`)

This also matches `src/PersonObjects/Person.ts` structure, but will be script breaking

**Comment by catloversg** [comment]
@d0sboots Can we close this issue? I don't think we will (ever) change common property identifiers (e.g., `hack`) or commonly used functions (e.g., `ns.hack`). This type of issue (unnecessary RAM cost) is just unavoidable with how the static RAM calculator works. It can be mitigated to a moderate extent by Snarling's new proposal, though.

**Comment by d0sboots** [comment]
I'm leaving it open as a tracking/collection bug for this type of issue. As you say, it won't be fixed, but it *is* still an issue, and it's good to have a single place to dupe them all to.

**Comment by d0sboots** [comment]
Important general info:
There are actually two RAM usages, the static limit and the dynamic limit.

The static RAM limit is what is calculated by static analysis and reported in the green box in the editor. By *default*, it is how much RAM your script will take up when run, and it is what is being fooled by various objects or functions that are named the same as ns APIs.

The dynamic RAM limit is the "true" amount of RAM that your script needs to run. It is calculated by looking at all the ns functions that your script calls over its life. If you have a call in your script, but never end up executing it, it won't count towards the dynamic limit. If the dynamic limit ever exceeds the allocated RAM size of your script, it will be aborted immediately.

Most players never need to worry or care about the dynamic limit, because the static limit is conservative and ensures that enough RAM is always allocated. However, advanced strategies can be employed to save on RAM by knowing about the difference. Also, there are a number of methods available both for advanced strategies, and to adjust your static RAM allocation if you are being hit by static analysis bugs.

Some mitigations that are worth mentioning on this master bug:
* You can use string index style (`object["hack"]`) to avoid it being picked up by the RAM analysis.
* You can use `--ram-override` when running scripts on the commandline to explicitly provide the RAM usage of your script.
* You can use `ramOverride` in [RunOptions](https://github.com/bitburner-official/bitburner-src/blob/dev/markdown/bitburner.runoptions.md) to explicitly provide the RAM usage when using `ns.run` and similar calls.
* You can use the `ns.ramOverride` function to change the per-thread static RAM limit of a running script.
* Using `ns.ramOverride` as the first function in `main()`, with a constant value, will also set the *static* RAM limit for your script. (You will see it reflected in the editor as "override (misc)"

**Comment by natedogith1** [comment]
Would it be possible to add some way to tell the static checker that it's wrong?  I'm imagining a comment at the beginning of the script either with either the ram difference or the list of functions that should/shouldn't be counted.
```ts
// @ram-checker-add -0.4
```
```ts
// @ram-checker-ignore hack, weaken, grow
```

Or both for a more complex situation (e.g. script only calls one of weaken/grow)
```ts
// @ram-checker-ignore weaken, grow
// @ram-checker-add 0.15
```

**Comment by catloversg** [comment]
@d0sboots I remember that jeek suggested overriding RAM via a one-line command. Do you remember if the suggestion was vetoed or there was just not enough demand (especially after you added `ns.ramOverride`)?

If it's not vetoed, should we support only `@ram-override` or the fancy things suggested here?

**Comment by d0sboots** [comment]
I missed the conversation here, but you can already override RAM via a one-line command. This is the last of the ways that I mentioned:
> Using `ns.ramOverride` as the first function in `main()`, with a constant value, will also set the static RAM limit for your script. (You will see it reflected in the editor as "override (misc)"
This looks like:
```js
export async function main(ns) {
  ns.ramOverride(2.5);
  //...
```
I don't think the fancier add/subtract methods are worth it; we already have a lot of complexity in this space and a lot of tools for people to build their own, more complicated solutions if they want.

**Comment by fluxxdog** [comment]
Got another collision but it's a little different, didn't know if this should be spun off into its own or not.

This one is with a native JS function: [RegExp.exec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec)
